lib/sycamore/sycamore/query/client.py

```diff
@@ -100,6 +100,18 @@ class SycamoreQueryClient:
         os_config (optional): OpenSearch configuration. Defaults to DEFAULT_OS_CONFIG.
         os_client_args (optional): OpenSearch client arguments. Defaults to DEFAULT_OS_CLIENT_ARGS.
         trace_dir (optional): Directory to write query execution trace.
+
+    Notes:
+        If you override the context. You cannot override the s3_cache_path or os_client_args; you need
```

Comment:
This seems a little confusing. Maybe we should just get rid of these parameters and require a custom context being passed in if any of these parameters needs to be modified.

Reply:
I agree, but I didn't want to break existing code.